spl-balance/src/option.rs

```diff
@@ -0,0 +1,951 @@
+//! A C representation of Rust's `std::option::Option` used across the FFI
+//! boundary for Solana program interfaces
+//!
+//! This implementation mostly matches `std::option` except iterators since the iteration
+//! trait requires returning `std::option::Option`
+
+use std::{
+    convert, mem,
+    ops::{Deref, DerefMut},
+};
+
+/// A C representation of Rust's `std::option::Option`
+#[repr(C)]
+#[derive(Copy, PartialEq, PartialOrd, Eq, Ord, Debug, Hash)]
+pub enum COption<T> {
```

Comment:
This whole file can be removed and you can switch back to using std::Option instead